### PR TITLE
Improve centos7 osd runner to correctly export fpm and allows dynamic node version switch

### LINKED ISSUE #1887: RPM build for core dashboards failed in fpm assemble step
When running RPM build for OSD 1.3.0 from infra's docker container, the following error shows up. (sample [run](http://jenki-jenki-fpgmrv2ryxko-1366042710.us-east-1.elb.amazonaws.com/job/Playground/job/tianleh-test/job/tianleh-osd-build-3-22/13/execution/node/108/log/))

The input manifest is https://github.com/opensearch-project/opensearch-build/blob/main/manifests/1.3.0/opensearch-dashboards-1.3.0.yml
The command is `Running ./src/run_build.py manifests/1.3.0/opensearch-dashboards-1.3.0.yml -d rpm -p linux -a x64 ...`

```

│ info $ fpm --force -t rpm --package /tmp/tmp8l74j355/OpenSearch-Dashboards/target/NAME-1.3.0-ARCH.TYPE -s dir --name opensearch-dashboards --description Explore and visualize your OpenSearch data --version 1.3.0 --url https://opensearch.org/ --vendor OpenSearch --maintainer OpenSearch Dashboards Team <opensearch@amazon.com> --license ASL 2.0 --conflicts opensearch-dashboards --after-install /tmp/tmp8l74j355/OpenSearch-Dashboards/src/dev/build/tasks/os_packages/package_scripts/post_install.sh --before-install /tmp/tmp8l74j355/OpenSearch-Dashboards/src/dev/build/tasks/os_packages/package_scripts/pre_install.sh --before-remove /tmp/tmp8l74j355/OpenSearch-Dashboards/src/dev/build/tasks/os_packages/package_scripts/pre_remove.sh --after-remove /tmp/tmp8l74j355/OpenSearch-Dashboards/src/dev/build/tasks/os_packages/package_scripts/post_remove.sh --config-files /etc/opensearch-dashboards/opensearch_dashboards.yml --template-value user=opensearch-dashboards --template-value group=opensearch-dashboards --template-value optimizeDir=/usr/share/opensearch-dashboards/optimize --template-value configDir=/etc/opensearch-dashboards --template-value pluginsDir=/usr/share/opensearch-dashboards/plugins --template-value dataDir=/var/lib/opensearch-dashboards --exclude usr/share/opensearch-dashboards/config --exclude usr/share/opensearch-dashboards/data --exclude usr/share/opensearch-dashboards/assets --architecture x64 --rpm-os linux /tmp/tmp8l74j355/OpenSearch-Dashboards/build/opensearch-dashboards-1.3.0-linux-x64/=/usr/share/opensearch-dashboards/ /tmp/tmp8l74j355/OpenSearch-Dashboards/build/opensearch-dashboards-1.3.0-linux-x64/config/=/etc/opensearch-dashboards/ /tmp/tmp8l74j355/OpenSearch-Dashboards/build/opensearch-dashboards-1.3.0-linux-x64/data/=/var/lib/opensearch-dashboards/ /tmp/tmp8l74j355/OpenSearch-Dashboards/src/dev/build/tasks/os_packages/service_templates/sysv/=/ /tmp/tmp8l74j355/OpenSearch-Dashboards/src/dev/build/tasks/os_packages/service_templates/systemd/=/
   │ERROR failure 0 sec
   │ERROR Error [ERR_STREAM_PREMATURE_CLOSE]: Premature close
   │          at Socket.onclose (internal/streams/end-of-stream.js:52:15)
   │          at Socket.emit (events.js:203:15)
   │          at Pipe._handle.close (net.js:607:12)

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
2022-03-31 16:58:57 ERROR    Error building OpenSearch-Dashboards, retry with: ./build.sh manifests/1.3.0/opensearch-dashboards-1.3.0.yml --component OpenSearch-Dashboards
Traceback (most recent call last):
  File "./src/run_build.py", line 80, in <module>
    sys.exit(main())
  File "./src/run_build.py", line 68, in main
    builder.build(build_recorder)
  File "/var/jenkins/workspace/Playground/tianleh-test/tianleh-osd-build-3-22@2/src/build_workflow/builder_from_source.py", line 55, in build
    self.git_repo.execute(build_command)
  File "/var/jenkins/workspace/Playground/tianleh-test/tianleh-osd-build-3-22@2/src/git/git_repository.py", line 83, in execute
    subprocess.check_call(command, cwd=cwd, shell=True)
  File "/usr/local/lib/python3.7/subprocess.py", line 363, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command 'bash /var/jenkins/workspace/Playground/tianleh-test/tianleh-osd-build-3-22@2/scripts/components/OpenSearch-Dashboards/build.sh -v 1.3.0 -p linux -a x64 -d rpm -s false -o builds' returned non-zero exit status 1.

```

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Improve centos7 osd runner to correctly export fpm and allows dynamic node version switch

1. The fpm utility now correctly exported to env, means OSD core build rpm will not crash anymore
2. Add an entrypoint to dynamically export node version when runner starts. If no params, use 10.24.1; if use `-e "NODE_VERSION=14.18.2"` it uses 14.18.2. We can add more versions easily later. This makes the dashboards builds the same way as opensearch since you can now apply different node versions for different releases in input manifest.

```
% docker run --rm -it -u root opensearchstaging/ci-runner:test123
v10.24.1 is already installed.
Now using node v10.24.1 (npm v6.14.12)
Now using node v10.24.1 (npm v6.14.12)
```

```
% docker run --rm -it -e "NODE_VERSION=14.18.2" opensearchstaging/ci-runner:test123
v14.18.2 is already installed.
Now using node v14.18.2 (npm v6.14.15)
Now using node v14.18.2 (npm v6.14.15)
```

Test image: https://hub.docker.com/r/opensearchstaging/ci-runner/tags?page=1&ordering=last_updated&name=ci-runner-centos7-opensearch-dashboards-build-v1

Thanks.

 
### Issues Resolved
#1887
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by tianleh** [comment]
I will test this change in my https://github.com/opensearch-project/opensearch-build/pull/1807 once the new tag is ready

File: docker/ci/dockerfiles/build.centos7.opensearch-dashboards.x64.arm64.dockerfile [hunk 1]
```diff
@@ -115,66 +102,43 @@ SHELL ["/bin/bash", "-lc"]
...
-    gem install fpm -v 1.13.0
+RUN . /etc/profile.d/rvm.sh && rvm install 2.4.0 && rvm --default use 2.4.0 && yum install -y rpm-build
 
 ENV RUBY_HOME=/usr/local/rvm/rubies/ruby-2.4.0/bin
 ENV RVM_HOME=/usr/local/rvm/bin
 ENV GEM_HOME=/usr/share/opensearch/.gem
 ENV GEM_PATH=$GEM_HOME
 ENV PATH=$RUBY_HOME:$RVM_HOME:$PATH
 
-# Install Python37 binary
```
Comment (tianleh) [inline docker/ci/dockerfiles/build.centos7.opensearch-dashboards.x64.arm64.dockerfile:128]: We still need python to run the build script

**Comment by tianleh** [comment]
I am performing testing in https://github.com/opensearch-project/opensearch-build/pull/1807 by using the new image generated by this PR. I will approve once my test is successful.

**Comment by tianleh** [comment]
Verified that the new image is working. (sample [run](http://jenki-jenki-fpgmrv2ryxko-1366042710.us-east-1.elb.amazonaws.com/job/Playground/job/tianleh-test/job/tianleh-osd-build-3-22/22/))
Screen Shot 2022-04-04 at 11 51 33 AM

**Review by tianleh — APPROVED** [review]

File: docker/ci/dockerfiles/build.centos7.opensearch-dashboards.x64.arm64.dockerfile (hunk 1 above)
Comment (peterzhuamazon) [reply docker/ci/dockerfiles/build.centos7.opensearch-dashboards.x64.arm64.dockerfile]: Added now, thanks.

**Pull request merged**